modules/fedimint-wallet-client/src/deposit.rs

```diff
@@ -249,15 +250,24 @@ async fn await_btc_transaction_confirmed(
 
         debug!(proof_block_hash = ?txout_proof.block_header.block_hash(), "Generated merkle proof");
 
-        return txout_proof;
+        let consensus_version = match global_context.module_api().module_consensus_version().await {
```

Comment:
This codepath is no longer used once we introduced the peg-in monitor in v0.4.0. It existed only for allowing clients that upgraded with active state machines to gracefully finish, so we should remove this logic.

This is more of an FYI and I don't believe this change is harmful.

Tracking the cleanup in https://github.com/fedimint/fedimint/issues/6675